app/v2/models/responses/hateoas/HateoasDepartureUpdateMovementResponse.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.models.responses.hateoas
+
+import play.api.libs.json.JsObject
+import play.api.libs.json.Json
+import v2.models.DepartureId
+import v2.models.MessageId
+
+object HateoasDepartureUpdateMovementResponse extends HateoasResponse {
+
+  def apply(departureId: DepartureId, messageId: MessageId): JsObject =
+    Json.obj(
+      "_links" -> Json.obj(
+        "self"     -> Json.obj("href" -> departureUri(departureId)),
+        "messages" -> Json.obj("href" -> messageUri(departureId, messageId))
```

Comment:
I hadn't realised you hadn't done this, anyway...

I think you've got this wrong here, `self` should be what `messages` is, and what you've put in `self` should be `departure`.

Reply:
Thanks Dan. Done the changes as suggested.